Buildings/Experimental/OpenBuildingControl/CDL/Logical/Composite/OnHold.mo

```diff
@@ -0,0 +1,102 @@
+within Buildings.Experimental.OpenBuildingControl.CDL.Logical.Composite;
+model OnHold "Block that holds a signal on for a requested time period"
+
+  LessThreshold                                               les1(threshold=
+        holdOnDuration)
+    annotation (Placement(transformation(extent={{-20,-40},{0,-20}})));
+  Continuous.Constant Zero(final k=0)
+    annotation (Placement(transformation(extent={{-100,-40},{-80,-20}})));
+  Timer                                                        timer
+    annotation (Placement(transformation(extent={{20,10},{40,30}})));
+  Modelica.Blocks.Logical.Pre pre
+    annotation (Placement(transformation(extent={{50,40},{70,60}})));
+  Not not1
+    annotation (Placement(transformation(extent={{-60,40},{-40,60}})));
+  Equal                                                        equ1
+    annotation (Placement(transformation(extent={{-70,-40},{-50,-20}})));
+  Or or2 annotation (Placement(transformation(extent={{-20,60},{0,80}})));
+  And and2 annotation (Placement(transformation(extent={{20,40},{40,60}})));
+  Interfaces.BooleanOutput                                                y
+    annotation (Placement(transformation(extent={{100,-10},{120,10}}),
+        iconTransformation(extent={{100,-10},{120,10}})));
+  Interfaces.BooleanInput u
+    annotation (Placement(transformation(extent={{-140,-20},{-100,20}}),
+        iconTransformation(extent={{-120,-10},{-100,10}})));
+  parameter Real holdOnDuration(unit="s") = 3600 "Time duration of the ON hold.";
```

Comment:
Use Modelica.SIunits.Time and then the `unit` attribute need not be set.
Also, list first all (public) parameters, then all input signals, then all output signals, and then the instances of variables and blocks.